databricks/koalas/frame.py

```diff
@@ -3742,6 +3742,127 @@ def drop_duplicates(self, subset=None, inplace=False):
+        -------
+        DataFrame
+            Unpivoted DataFrame.
+
+        Examples
+        --------
+        >>> df = ks.DataFrame({'A': {0: 'a', 1: 'b', 2: 'c'},
+        ...                    'B': {0: 1, 1: 3, 2: 5},
+        ...                    'C': {0: 2, 1: 4, 2: 6}})
+        >>> df
+           A  B  C
+        0  a  1  2
+        1  b  3  4
+        2  c  5  6
+
+        >>> ks.melt(df)  # doctest: +ELLIPSIS +NORMALIZE_WHITESPACE
+            variable value
+        ...        A     a
+        ...        B     1
+        ...        C     2
+        ...        A     b
+        ...        B     3
+        ...        C     4
+        ...        A     c
+        ...        B     5
+        ...        C     6
+
+        >>> df.melt(id_vars='A')  # doctest: +ELLIPSIS +NORMALIZE_WHITESPACE
+             A variable  value
+        ...  a        B      1
+        ...  a        C      2
+        ...  b        B      3
+        ...  b        C      4
+        ...  c        B      5
+        ...  c        C      6
+
+        >>> ks.melt(df, id_vars=['A', 'B'])  # doctest: +ELLIPSIS +NORMALIZE_WHITESPACE
+             A  B variable  value
+        ...  a  1        C      2
+        ...  b  3        C      4
+        ...  c  5        C      6
+
+        >>> df.melt(id_vars=['A'], value_vars=['C'])  # doctest: +ELLIPSIS +NORMALIZE_WHITESPACE
+             A variable  value
+        ...  a        C      2
+        ...  b        C      4
+        ...  c        C      6
+
+        The names of 'variable' and 'value' columns can be customized:
+
+        >>> ks.melt(df, id_vars=['A'], value_vars=['B'],
+        ...         var_name='myVarname', value_name='myValname')
+        ... # doctest: +ELLIPSIS +NORMALIZE_WHITESPACE
+             A myVarname  myValname
+        ...  a         B          1
+        ...  b         B          3
+        ...  c         B          5
+        """
+        if id_vars is None:
+            id_vars = []
+        if not isinstance(id_vars, (list, tuple, np.ndarray)):
+            id_vars = list(id_vars)
+
+        data_columns = self._metadata.data_columns
+
+        if value_vars is None:
+            value_vars = []
+        if not isinstance(value_vars, (list, tuple, np.ndarray)):
+            value_vars = list(value_vars)
+        if len(value_vars) == 0:
+            value_vars = data_columns
+
+        data_columns = [data_column for data_column in data_columns if data_column not in id_vars]
+        sdf = self._sdf
+
+        pairs = F.explode(F.array(*[
+            F.struct(*(
+                [F.lit(column).alias(var_name)] +
+                [F.col(id_var) for id_var in id_vars] +
+                [F.col(column).alias(value_name)])
+            ) for column in data_columns if column in value_vars]))
+
+        # New index column is always single index.
+        internal_index_column = "__index_level_0__"
+
+        columns = ([F.monotonically_increasing_id().alias(internal_index_column)] +
```

Comment:
one thing different from Pandas is that we use `monotonically_increasing_id` instead of sequential id

Reply:
Hmm .. should we just don't set the index? I am not sure which way is better. Pandas' looks setting range index for this case

Reply:
So far, we just don't set the index if pandas resets and sets range index.
Maybe we need a discussion about such a case, whether we should use `monotonically_increasing_id` or leave it as-is.

Reply:
Hm .. okie. let me remove it here for now.